komorebi-bar/src/widget.rs

```diff
@@ -21,9 +22,22 @@ use eframe::egui::Ui;
 use schemars::JsonSchema;
 use serde::Deserialize;
 use serde::Serialize;
+use std::sync::Arc;
 
 pub trait BarWidget {
-    fn render(&mut self, ctx: &Context, ui: &mut Ui);
+    fn render(&mut self, ctx: &Context, ui: &mut Ui, config: RenderConfig);
+}
+
+#[derive(Clone)]
+pub struct RenderConfig {
+    /// Sets how widgets are grouped
+    _group: i32,
```

Comment:
For this I think an enum like `Grouping` would be nice:

```rust
enum Grouping {
    None,
    Rounded,
    // others in the future?
}
```

Then inside the widget's render implementation we can go through and set the current defaults to match on `None`, and each new grouping strategy can be implemented in its own match variant arm.

Reply:
Thank you for helping, I did a bit more on this. I am thinking about having at least 2 groupings. I will chip away the work on those in my free time :)

Reply:
@LGUG2Z I think I have most of it figured out, just need to add more grouping. I will resolve this for now